### DISCUSSION TITLE
4.x 支付回调handlePaidNotify如何处理多个商户号配置的情况？

### DISCUSSION THREAD

**Discussion by zhuman90** [w7corp/easywechat#2803] [General]
文档上写的是直接$response = $app->handlePaidNotify(fn)，但是我的$app是需要通过配置参数生成的：$app = Factory::payment($config);

我现在有多个商户号，通过订单能获取到，在支付回调的时候，我怎么来确定用哪个$config来创建$app？我直接自己解析request，然后拿到out_trade_no？那这样的话，handlePaidNotify不就没啥用了

**Comment by overtrue** [comment]
你应该用 url 区分🥹

**Reply by zhuman90** [reply]
您的意思是，不同的商户号支付的时候回调地址设置成不同的吗？

**Reply by overtrue** [reply]
对，因为即便是我给你提供了微信的回调内容，也是加密的

**Reply by overtrue** [reply]
你也可以直接 new server 后拿到推送的内容的：

> https://github.com/w7corp/easywechat/blob/6.x/src/Pay/Server.php#L114

```php
$server->getRequestMessage();
```

**Reply by overtrue** [reply]
不过 new server 也需要 merchant id

**Reply by overtrue** [reply]
还是建议从回调 URL 上下功夫

**Reply by zhuman90** [reply]
好的，非常感谢，我去研究研究
